crates/bellman/src/plonk/polynomials/mod.rs

```diff
@@ -1272,27 +1276,14 @@ impl<F: PrimeField> Polynomial<F, Values> {
 
         assert!(subset_factor.is_power_of_two());
 
-        let current_size = self.coeffs.len();
-        let new_size = current_size / subset_factor;
-
-        let mut result = Vec::with_capacity(new_size);
-        unsafe { result.set_len(new_size) };
-
         // copy elements. If factor is 2 then non-reversed we would output only elements that are == 0 mod 2
         // If factor is 2 and we are bit-reversed - we need to only output first half of the coefficients
         // If factor is 4 then we need to output only the first 4th part
         // if factor is 8 - only the first 8th part
+        let current_size = self.coeffs.len();
+        let new_size = current_size / subset_factor;
 
-        let start = 0;
-        let end = new_size;
-
-        result.copy_from_slice(&self.coeffs[start..end]);
-
-        // unsafe { result.set_len(new_size)};
-        // let copy_to_start_pointer: *mut F = result[..].as_mut_ptr();
-        // let copy_from_start_pointer: *const F = self.coeffs[start..end].as_ptr();
-
-        // unsafe { std::ptr::copy_nonoverlapping(copy_from_start_pointer, copy_to_start_pointer, new_size) };
+        let result = self.coeffs[..new_size].to_vec();
```

Comment:
Could this worsen benchmark results?

Reply:
It should not. Maybe it'll even be better. Previously we allocated memory and copied data there as 2 steps, but now we do it as a single step. 99% that both versions compile to the same assembly, but idk if rust has some optimizations for `slice::to_vec`.